source/features/close-out-of-view-modals.tsx

```diff
@@ -1,29 +1,58 @@
+import select from 'select-dom';
+import onetime from 'onetime';
+import delegate from 'delegate-it';
+
 import features from '.';
 
-const observer = new IntersectionObserver(([{intersectionRatio, target}]) => {
-	if (intersectionRatio === 0) {
-		observer.unobserve(target);
-		target.closest('details')!.open = false;
+const visible = new Set();
+const observer = new IntersectionObserver(entries => {
+	let lastModal: Element;
+	for (const {intersectionRatio, target: modal} of entries) {
+		if (intersectionRatio > 0) {
+			visible.add(modal);
+		} else {
+			visible.delete(modal);
+		}
+
+		lastModal = modal;
+	}
+
+	if (visible.size === 0) {
+		observer.disconnect();
+		lastModal!.closest('details')!.open = false;
 	}
 });
 
+let lastOpen: number;
+let delegation: delegate.Subscription;
 function menuActivatedHandler(event: CustomEvent): void {
-	const details = event.target as HTMLElement;
-	const modalBox = details.querySelector('details-menu')!;
-
-	// Avoid silently breaking the interface: #2701
-	if (modalBox.getBoundingClientRect().width === 0) {
-		features.error(__filebasename, 'Modal element was not correctly detected for', details);
-	} else {
-		observer.observe(modalBox);
+	const details = event.target as HTMLDetailsElement;
+
+	// Safety check #3742
+	if (!details.open && lastOpen > Date.now() - 1000) {
+		delegation!.destroy();
+		features.error(__filebasename, 'Modal was closed too quickly. Disabling feature');
```

Comment:
The drawback is that if the user is _Flash_, they will see this error in the console. 🤷‍♂️